pkg/compact/compact.go

```diff
@@ -1008,12 +1018,23 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 			uniqueSources[s] = struct{}{}
 		}
 
-		if err := block.Download(ctx, cg.logger, cg.bkt, meta.ULID, bdir); err != nil {
+		// creating a child block span's context, identified by the block ID.
+		blockSpan, blockCtx := tracing.StartSpan(ctx, "block_operations", opentracing.Tags{"block ID": meta.ULID})
+		defer blockSpan.Finish()
```

Comment:
This seems incorrect. Using defer will close this span at the end of the function. Btw I guess we don't need a separate span for `block_operations`. WDYT?

Reply:
I thought to minimise the number of spans at each level, it would be better to have a block span with child spans for each block op(like I showed you in our 1:1).   Then to trace a block's trajectory, we just need to search using block ID as the tag/attribute.
I was considering adding other block operations like delete to the span too.  Something like:  
1. Block span:
a. Block download
b. Healthcheck
c. Upload
d. Delete. 

Does this reasoning tie in with general tracing principles/best practices?

Reply:
Re: cleaning up the PR: I think, apart from the review changes, the only irrelevant changes I've added are comments for my understanding?
(I'll remove those in my upstream PR)

Reply:
This is not addressed. You are still using a defer in a for loop.

Reply:
I was waiting till we reached consensus on separate/combined spans for each block. 
Do you think separate spans for each op are a good idea, or a combined span for a block?

Reply:
> Then to trace a block's trajectory, we just need to search using block ID as the tag/attribute.

That's another thing. You can have block ID as tag/attribute without using a separate block span, right? Then what's the point of using the parent `block_operation` span? It is meaningless to me

Reply:
> I thought to minimise the number of spans at each level

This is okay because it is easier to view but the side affect is that you will get more spans with no actual meaning.

Reply:
You can keep this span first span and let's wait for other maintainer's review when you submit an upstream pr. Let's first fix the defer issue. That's a bug no doubt.

Reply:
Alright, I've moved the `defer`.